src/DotVVM.Framework/Runtime/Commands/EventValidator.cs

```diff
@@ -99,54 +146,121 @@ private FindBindingResult FindControlCommandBinding(string[] path, string comman
+                        if (currentValidationTargetPath != validationTargetPath)
+                        {
+                            exceptionPropertyKey = exceptionPropertyKey == null
+                                ? "binding id"
+                                : string.Join(", ", exceptionPropertyKey, "validation path");
+                            infoMessage.Append(
+                                $"Expected validation path: '{string.Join("/", validationTargetPath)}' Command binding validation path: '{string.Join("/", currentValidationTargetPath)}'");
+                        }
+
+                        //checking if binding is control binding
+                        isControl = control.GetClosestControlBindingTarget() == targetControl;
+
+                        if (exceptionPropertyKey == null && isControl)
+                        {
+                            //correct binding found
+                            resultBinding = (CommandBindingExpression)binding.Value;
+                            resultControl = control;
+                            resultProperty = binding.Key;
+                        }
+                        else if (exceptionPropertyKey != null)
+                        {
+                            exceptionPropertyKey = (isControl ? "Control command bindings with wrong " : "Command bindings with wrong ") + exceptionPropertyKey;
+                            if (!candidateBindings.ContainsKey(exceptionPropertyKey))
                             {
-                                // the validation path is equal, we have found the binding
-                                resultBinding = (CommandBindingExpression)binding.Value;
-                                resultProperty = binding.Key;
-                                resultControl = control;
+                                candidateBindings.Add(exceptionPropertyKey, new CandidateBindings());
                             }
+                            candidateBindings[exceptionPropertyKey]
+                                .AddBinding(new KeyValuePair<string, IBinding>(infoMessage.ToString(), binding.Value));
+                        }
+                        else
+                        {
+                            errorMessage = "Invalid command invocation (the binding is not control command binding)";
                         }
                     }
                 }
             });
 
             return new FindBindingResult
             {
-                Property = resultProperty,
+                ErrorMessage = bindingInPath ? errorMessage : "Nothing was found in found inside specified DataContext, check if ViewModel is populated",
+                CandidateBindings = candidateBindings,
                 Binding = resultBinding,
-                Control = resultControl
+                Control = resultControl,
+                Property = resultProperty
```

Comment:
That's a possibility, but I think that the `FindControlCommandBinding` method with `targetControl` parameter set to null will behave the same as `FindCommandBinding` method ;) Or maybe just a minor tweak would be required.